test/rubygems/test_gem.rb

```diff
@@ -29,6 +29,17 @@ def setup
     util_remove_interrupt_command
   end
 
+  def teardown
+    super
+
+    change_version @version if defined? @version
```

Comment:
Great point. I felt at a loss of how to "fake this" global state, which is why I landed on this (no ideal) solution. Do you have a suggestions for updating the version without reaching into `Gem` and modifying an instance variable?